File: src/main/java/no/echokarriere/backend/configuration/CrudRepository.java
package no.echokarriere.backend.configuration;

import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

public interface CrudRepository<E, I> {
    @Transactional
    Optional<E> create(E entity);

    @Transactional
    Optional<E> update(E entity);

    @Transactional
    boolean delete(I id);

    @Transactional
    Optional<E> select(I id);

    @Transactional
    List<E> selectAll();
}

File: src/main/kotlin/no/echokarriere/configuration/CrudRepository.kt
package no.echokarriere.configuration

interface CrudRepository<E, I> {
    suspend fun selectAll(): List<E>
    suspend fun select(id: I): E?
    suspend fun insert(entity: E): E?
    suspend fun update(entity: E): E?
    suspend fun delete(id: I): Boolean
}
